src/analyzer.ts

```diff
@@ -191,26 +192,33 @@ const getDefinitelyUsed = (file: SourceFile): IAnalysedResult[] => ([
   ...exportWildCards(file),
 ]);
 
-export const getPotentiallyUnused = (file: SourceFile): IAnalysedResult => {
+export const getPotentiallyUnused = (file: SourceFile, skipPattern?: string): IAnalysedResult => {
   const exported = getExported(file);
 
   const idsInFile = file.getDescendantsOfKind(ts.SyntaxKind.Identifier);
   const referenceCounts = countBy(x => x)((idsInFile || []).map(node => node.getText()));
   const referencedInFile = Object.entries(referenceCounts)
     .reduce(
-      (previous, [name, count]) => previous.concat(count > 1 ? [name] : []), 
+      (previous, [name, count]) => previous.concat(count > 1 ? [name] : []),
       []
     );
 
-  const referenced = file
-    .getReferencingNodesInOtherSourceFiles()
-    .reduce(
+  let references = file.getReferencingNodesInOtherSourceFiles();
```

Comment:
Could you kindly refactor this to be a function, so we don't have conditional logic within this function, and eliminate the `let`?

Something like:

```ts
const getReferences = (originalList, regex) => {
  if (skipPattern) {
    return ...
  } else {
    return originalList;
  }
}
```

So that this section becomes:

```ts
const referenced = getReferences(file.getReferencingNodesInOtherSourceFiles(), skipPattern).reduce(...)
```

Reply:
Haha, I actually considered this! Yeah, absolutely.